iceoryx_dds/include/iceoryx_dds/dds/cyclone_data_reader.hpp

```diff
@@ -27,34 +27,60 @@ namespace iox
 {
 namespace dds
 {
-///
 /// @brief Implementation of the DataReader abstraction using the cyclonedds implementation.
-///
 class CycloneDataReader : public DataReader
 {
   public:
     CycloneDataReader() = delete;
+
+    /// @brief Constructor to set cyclone data reader object from given IDs
+    /// @param[in] serviceId ID of the service
+    /// @param[in] instanceId ID of the instance of the service
+    /// @param[in] eventId ID of the event
     CycloneDataReader(const capro::IdString_t serviceId,
                       const capro::IdString_t instanceId,
                       const capro::IdString_t eventId) noexcept;
+
     virtual ~CycloneDataReader();
 
     CycloneDataReader(const CycloneDataReader&) = delete;
     CycloneDataReader& operator=(const CycloneDataReader&) = delete;
     CycloneDataReader(CycloneDataReader&&) = delete;
     CycloneDataReader& operator=(CycloneDataReader&&) = delete;
 
+    /// @brief Connect cylcone data reader to the underlying DDS network
     void connect() noexcept override;
 
+    /// @brief Get the size of the next sample if one is available
+    /// @return an optional pointer to the size of the next sample if one is available, cxx::nullopt_t if value is not
+    /// initialized
     iox::cxx::optional<uint32_t> peekNextSize() override;
+
+    /// @brief Check if new samples are ready to take
+    /// @return true if there are samples available, otherwise false
     bool hasSamples() override;
+
+    /// @brief Take the next available sample from the DDS data space.
+    /// @param[in] buffer Receive buffer in which sample will be stored.
+    /// @param[in] bufferSize Size of the provided buffer.
```

Comment:
please don't repeat yourself and have it just `@param[in] buffer to store the received sample` and `@param[in] bufferSize of provided buffer`

Please also adjust it at on the other methods which repeat the variable name

Reply:
Move to base class and fixed.